pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -231,7 +231,10 @@ public ServerCnx(PulsarService pulsar) {
     }
 
     public ServerCnx(PulsarService pulsar, String listenerName) {
-        super(pulsar.getBrokerService().getKeepAliveIntervalSeconds(), TimeUnit.SECONDS);
+        // pulsar.getBrokerService() can sometimes be null in unit tests when using mocks
+        // the null check is a workaround for #13620
+        super(pulsar.getBrokerService() != null ? pulsar.getBrokerService().getKeepAliveIntervalSeconds() : 0,
```

Comment:
is there a way to write a warning that we are using a default value ?

Reply:
I doubt that it's necessary. `pulsar.getBrokerService()` is never really null.  This is a special case caused by the mocking framework.

Reply:
Adding logging would require extracting a static method, that's why I avoided it to keep the workaround minimal.